tla/consistency/SingleNode.tla

```diff
@@ -63,11 +62,11 @@ RwTxExecuteAction ==
         \* that it can be picked up by the current leader or any former leader
         /\ \E view \in DOMAIN ledgerBranches:
                 ledgerBranches' = [ledgerBranches EXCEPT ![view] = 
-                    Append(@,[view |-> view, tx |-> history[i].tx])]
+                    @ @@ (Max(DOMAIN ledgerBranches[view] \cup {0})+1) :> [view |-> view, tx |-> history[i].tx]]
```

Comment:
We have this pattern in multiple places, can we pull it out and name it?

```
SparseLen(s) == Max(DOMAIN s \cup {0})

SparseAppend(s, e) == s @@ (SparseLen(s) :> e)
```

For the others that don't have multiple callers its less clear-cut, but maybe still more readable to introduce `SparseSelectSeq` and `SparseSubSeq`?